pkg/kubelet/cm/topologymanager/bitmask/bitmask.go

```diff
@@ -140,7 +141,13 @@ func (s *bitMask) IsNarrowerThan(mask BitMask) bool {
 
 // String converts mask to string
 func (s *bitMask) String() string {
-	return fmt.Sprintf("%064b", *s)
+	grouping := 2
```

Comment:
To confirm my understanding.... is this diff basically printing in groups of 2 until it hits all 0s, and then it stops?

To you know if this type of handling of printing bitmasks is common?

Reply:
It's not printing in groups of 2, per se. It's limiting the total number of bits displayed, based on the value `*s`(which is a 64 bit integer).

Unrolled It is equivalent to:
```
if *s > (1 << 62)
    return fmt.Sprintf("%064b", *s)
if *s > (1 << 60)
    return fmt.Sprintf("%062b", *s)
if *s > (1 << 58)
    return fmt.Sprintf("%060b", *s)
...
if *s > (1 << 4)
    return fmt.Sprintf("%006b", *s)
if *s > (1 << 2)
    return fmt.Sprintf("%004b", *s)
return fmt.Sprintf("%002b", *s)
```

So you may see:
```
01
10
0100
1000
```
but you will never see:
```
0
1
0001
100
```